db/migrate/20200226183018_create_calendar_events.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class CreateCalendarEvents < ActiveRecord::Migration[5.2]
+  def change
+    create_table :calendar_events do |t|
+      t.integer :topic_id, null: false
+      t.integer :post_id
+      t.integer :post_number
+      t.integer :user_id
+      t.string :username
+      t.string :description
+      t.datetime :start_date, null: false
+      t.datetime :end_date
+      t.string :recurrence
+      t.string :region
+      t.timestamps
+
+      t.index :topic_id
+      t.index :post_id
+      t.index :user_id
+    end
+
+    # Rebuild calendar events
+    calendar_topic_ids = Post
```

Comment:
The following should be SQL for future compatibility.

Reply:
I agree with you here, but building a SQL is not a trivial task because calendar events used to be stored in OP only and now I am moving them in a `CalendarEvent` which belongs to the post where the event was extracted from.

Reply:
The thing is we have to do something, because if you leave a reference to a model in here and the structure of that model ever changes the old migration will break. By putting a reference to a model in a migration means the migration will never run unless that model exists with the exact same structure!

One thing you could do is convert the 1st and 3rd queries to SQL. Then for the second query that loops through posts, surround the whole thing with a rescue `nil`. Normally I would not recommend swallowing errors but in the case that the migration fails it's better to not have any calendar info than break the ability to run migrations at all.